src/main/java/edu/cornell/kfs/module/ar/document/validation/impl/CuTotalAmountBilledToDateExceedsAwardTotalSuspensionCategory.java

```diff
@@ -0,0 +1,59 @@
+package edu.cornell.kfs.module.ar.document.validation.impl;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kuali.kfs.coreservice.framework.parameter.ParameterService;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.module.ar.ArConstants;
+import org.kuali.kfs.module.ar.document.ContractsGrantsInvoiceDocument;
+import org.kuali.kfs.module.ar.document.validation.impl.TotalAmountBilledToDateExceedsAwardTotalSuspensionCategory;
+import org.kuali.kfs.module.cg.businessobject.Award;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+
+import edu.cornell.kfs.module.ar.CuArConstants;
+import edu.cornell.kfs.module.cg.businessobject.AwardExtendedAttribute;
+
+public class CuTotalAmountBilledToDateExceedsAwardTotalSuspensionCategory extends TotalAmountBilledToDateExceedsAwardTotalSuspensionCategory {
+    private static final Logger LOG = LogManager.getLogger(CuTotalAmountBilledToDateExceedsAwardTotalSuspensionCategory.class);
+    
+    protected transient ParameterService parameterService;
+    
+    @Override
+    public boolean shouldSuspend(ContractsGrantsInvoiceDocument contractsGrantsInvoiceDocument) {
+        if (shouldValidateOnBudgetTotal()) {
+            LOG.debug("shouldSuspend, validating based on award budget total");
+            KualiDecimal totalAmountBilledToDate = contractsGrantsInvoiceDocument.getInvoiceGeneralDetail().getTotalAmountBilledToDate();
+            
+            Award award = (Award) contractsGrantsInvoiceDocument.getInvoiceGeneralDetail().getAward();
+            AwardExtendedAttribute awardExtension = (AwardExtendedAttribute) award.getExtension();
+            KualiDecimal budgetTotalAmount = awardExtension.getBudgetTotalAmount();
+            if (ObjectUtils.isNull(budgetTotalAmount)) {
+                LOG.error("shouldSuspend, no budget amount set, setting budget amount to 0, wich will cause a suspension.");
```

Comment:
Minor typo in error message; should be "which" instead of "wich"

Reply:
ok